javascript/code-challenges/challenges-08.test.js

```diff
@@ -0,0 +1,333 @@
+    spouse: 'Khal Drogo',
+    children: ['Drogon', 'Rhaegal', 'Viserion'],
+    house: 'Targaryen'
+  },
+  {
+    name: 'Mace',
+    spouse: 'Alerie',
+    children: ['Margaery', 'Loras'],
+    house: 'Tyrell'
+  },
+  {
+    name: 'Euron',
+    spouse: null,
+    children: [],
+    house: 'Greyjoy'
+  },
+  {
+    name: 'Jon S.',
+    spouse: null,
+    children: [],
+    house: 'Snow'
+  }
+];
+
+const sortByChildren = (charArray) => {
+  // Solution code here...
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named containsW that takes in a string. This function should use a regular expression pattern to return true if the string contains the letter 'w' in lower case or false if it does not. 
+
+------------------------------------------------------------------------------------------------ */
+
+const containsW = (str) => {
+  // Solution code here...
+  let regex = /w/;
+  let answer = regex.test(str);
+  return answer;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named isNum that takes in a string or number of any length. This function should use a regular expression pattern to return true if the input contains a number, and false if the input does not contain a number.
+
+For example:
+12345 returns true
+'12345' returns true
+'h3llo world' returns true
+'hello world' returns false
+------------------------------------------------------------------------------------------------ */
+
+const isNum = (input) => {
+  // Solution code here...
+  let regex =/\d/;
+  let answer = regex.test(input);
+  return answer;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named containsWorld that takes in a string or number of any length. This function should use a regular expression pattern to return true if the input contains the word 'world' all in lower-case letters, and false if the input does not.
+
+------------------------------------------------------------------------------------------------ */
+
+const containsWorld = (input) => {
+  // Solution code here...
+  let regex =/world/;
+  let answer = regex.test(input);
+  return answer;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named isCapitalized that takes in a string. This function should use a regular expression pattern to match all words that begin with a capital letter. It should only match words, not punctuation.
+
+Return an array containing all the matches.
+------------------------------------------------------------------------------------------------ */
+
+const isCapitalized = (str) => {
+  // Solution code here...
+  // let regex = /[A-Z][a-z]*/g;
+  // let matched = str.match(regex);
+  // return matched;
+};
+
```

Comment:
Hi there, good work so far!

- Yes this is very close 👍 
- for the `*` can we use `\w` instead to specify only word characters or something perhaps, then keep the `/g`?
` /[A-Z]\w+/g`